src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,387 @@
+                    let info = self.breadcrumbs.borrow().get(index).map(|breadcrumb| {
+                        let definition = breadcrumb.info.method_pointer.clone();
+                        let call       = breadcrumb.info.expression_id;
+                        LocalCall{call,definition}
+                    }).as_ref().cloned();
+                    if info.is_some() {
+                        local_calls.push(info);
+                    } else {
+                        error!(self.logger, "LocalCall info is not present.");
+                        self.remove_breadcrumbs_history_beginning_from(index);
+                        break;
+                    }
+                }
+            },
+            cmp::Ordering::Equal => ()
+        }
+        info!(self.logger,"Selecting breadcrumb #{index}.");
+        (popped_count,local_calls)
+    }
+
+    fn push_breadcrumb(&self, local_call:&Option<LocalCall>) -> Option<(usize,usize)> {
+        if let Some(local_call) = local_call {
+            let method_pointer = &local_call.definition;
+            let expression_id  = &local_call.call;
+            let old_index      = self.current_index.get();
+            let new_index      = old_index + 1;
+
+            let breadcrumb_exists =
+                self.breadcrumbs.borrow_mut().get(old_index).contains_if(|breadcrumb| {
+                    breadcrumb.info.expression_id == *expression_id
+                });
+
+            if breadcrumb_exists {
+                debug!(self.logger, "Entering an existing {method_pointer.name} breadcrumb.");
+            } else {
+                debug!(self.logger, "Creating a new {method_pointer.name} breadcrumb.");
+                self.remove_breadcrumbs_history_beginning_from(self.current_index.get());
+                let breadcrumb       = Breadcrumb::new(&self.scene, method_pointer, expression_id);
+                let network          = &breadcrumb.frp.network;
+                let breadcrumb_index = new_index;
+                let model            = self.clone_ref();
+
+
+                // === User Interaction ===
+
+                frp::extend! { network
+                    _selected <- breadcrumb.frp.outputs.selected.map(move |_| {
+                        let (popped_count,local_calls) = model.select_breadcrumb(breadcrumb_index);
+                        for _ in 0..popped_count {
+                            model.frp_outputs.breadcrumb_pop.emit(());
+                        }
+                        for local_call in local_calls {
+                            model.frp_outputs.breadcrumb_push.emit(local_call);
+                        }
+                    });
+                }
+
+                info!(self.logger, "Pushing {breadcrumb.info.method_pointer.name} breadcrumb.");
+                breadcrumb.set_position(Vector3(self.width(),0.0,0.0));
+                self.breadcrumbs_container.add_child(&breadcrumb);
+                self.breadcrumbs.borrow_mut().push(breadcrumb);
```

Comment:
This leaks memory:
- `model` is `self.clone_ref()`
- `network` is a new FRP network owned by the `breadcrumb`.
- `_selected` is a FRP node in `network` which moves `model` inside, so it owns it.
- `model` contains `breadcrumbs` where we push `breadcrumb`, and we have a memory leak cycle.